openfl/interface/envoy.py

```diff
@@ -37,20 +37,37 @@ def envoy(context):
         help='The federation director port')
 @option('--tls/--disable-tls', default=True,
         is_flag=True, help='Use TLS or not (By default TLS is enabled)')
-@option('-sc', '--shard-config-path', default='shard_config.yaml',
-        help='The shard config path', type=ClickPath(exists=True))
+@option('-ec', '--envoy-config-path', default='envoy_config.yaml',
+        help='The envoy config path', type=ClickPath(exists=True))
 @option('-rc', '--root-cert-path', 'root_certificate', default=None,
         help='Path to a root CA cert')
 @option('-pk', '--private-key-path', 'private_key', default=None,
         help='Path to a private key')
 @option('-oc', '--public-cert-path', 'certificate', default=None,
         help='Path to a signed certificate')
-def start_(shard_name, director_host, director_port, tls, shard_config_path,
+def start_(shard_name, director_host, director_port, tls, envoy_config_path,
            root_certificate, private_key, certificate):
     """Start the Envoy."""
     logger.info('🧿 Starting the Envoy.')
-
-    shard_descriptor = shard_descriptor_from_config(shard_config_path)
+    # Reed the Envoy config
+    with open(envoy_config_path) as stream:
+        envoy_config = safe_load(stream)
+
+    envoy_params = envoy_config.get('params', {})
+    for plugin_name, plugin_settings in envoy_params.get('optional_plugin_components', {}).items():
+        template = plugin_settings.get('template')
+        if not template:
+            raise Exception('You should put a template'
+                            f'for plugin {plugin_name}')
+        class_name = template.split('.')[-1]
+        module_path = '.'.join(template.split('.')[:-1])
```

Comment:
This can be replaced with
```suggestion
        module_path, _, class_name = template.rpartition('.')
```

Reply:
Thank you